vent/io/hal.py

```diff
@@ -107,23 +110,48 @@ def secondary_pressure(self) -> float:
             raise RuntimeWarning('Secondary pressure sensor not instantiated. Check your "devices.ini" file.')
 
     @property
-    def flow(self) -> float:
-        """ The measured flow rate.
+    def flow_in(self) -> float:
+        """ The measured flow rate inspiratory side.
+        """
+        self._flow_sensor_in.update()
+        return self._flow_sensor_in.get()
+
+    @property
+    def flow_ex(self) -> float:
+        """ The measured flow rate expiratory side.
+        """
+        self._flow_sensor_en.update()
+        return self._flow_sensor_en.get()
+
+    @property
+    def setpoint_ex(self) -> float:
+        """ The currently requested flow on the expiratory side. This is solenoid open/close.
+
+        Returns:
+            float: 0<=setpoint<=1; The current set-point for flow control as a proportion of the maximum.
+        """
+        pass

```

Comment:
I can add this without a problem, although with our current setup it will be boolean open/close. Hope that's ok

Reply:
yeah, sure. The controller sets this to 0/1 anyway. I think you guys told me a while ago that this is an open/close valve.